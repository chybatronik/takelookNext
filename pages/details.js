import React, { Component } from 'react'
import * as actions from '../store/actions/shows'
import * as actionsEpisodes from '../store/actions/episodes'
import { Row, Col, Tag, Alert, Spin, Card, Table, Icon } from 'antd'
import { makeStore } from '../store'
import NavigateApp from '../components/Layout.js'
import withRedux from 'next-redux-wrapper'

const CheckableTag = Tag.CheckableTag

export class DetailsConatiner extends Component {
  static async getInitialProps ({query, store, isServer}) {
    console.log('getInitialProps DetailsConatiner')
    const state = store.getState()
    const { shows } = state.shows
    const show = shows[query.id]
    if (!show) {
      if (!state.shows.config.isFetch) {
        await store.dispatch(actions.getOneShow(query.id))
      }
    }

    const { episodes } = state.episodes
    const showEpisodes = episodes[query.id]
    if (!showEpisodes) {
      await store.dispatch(actionsEpisodes.getShowEpisodes(query.id))
    }
    return {...store.getState(), queryId: query.id, isServer}
  }

  componentDidMount () {
    window.scrollTo(0, 0)
  }

  renderLoader (config) {
    if (config.isError) {
      return (
        <NavigateApp title={'Ant Design'}>
          <Alert message='Ошибка получения данных с сервера' type='error' />
        </NavigateApp>
      )
    }
    return (
      <NavigateApp title={'Ant Design'}>
        <div className='spiner' key={'spiner'}>
          <Spin size='large' />
        </div>
      </NavigateApp>
    )
  }

  renderEpisodes () {
    const { episodes } = this.props.episodes
    const showEpisodes = episodes[this.props.queryId]
    if (!showEpisodes || showEpisodes.isFetch || showEpisodes.isError) {
      if (showEpisodes) {
        return this.renderLoader(showEpisodes)
      }
      return this.renderLoader({isFetch: true})
    }

    const dataSource = Object.keys(showEpisodes).map((id) => {
      const item = showEpisodes[id]
      return {
        name: `${item.season}x${item.number} ${item.name}`,
        age: item.airdate,
        address: item.url,
        key: item.id
      }
    })
    const columns = [{
      title: 'Episode Name',
      dataIndex: 'name',
      key: 'name'
    }, {
      title: 'Airdate',
      dataIndex: 'age',
      key: 'age'
    }, {
      title: 'Trailer',
      dataIndex: 'address',
      key: 'address',
      render: text => <a href={text}><Icon type='play-circle' /></a>
    }]
    return (
      <Table dataSource={dataSource} columns={columns} />
    )
  }

  render () {
    const { shows, config } = this.props.shows
    const show = shows[this.props.queryId]
    if (show) {
      return (
        <NavigateApp title={'Ant Design'}>
          <Row>
            <Col xs={{ span: 16, push: 8 }} sm={{ span: 16, push: 8 }} md={{ span: 16, push: 8 }} lg={{ span: 16, push: 8 }} >
              <h1>{show.name}</h1>
              <div style={{marginBottom: 20, marginTop: 10}}>
                {
                  show.genres.map(tag => (
                    <CheckableTag
                      key={tag}
                    >
                      {tag}
                    </CheckableTag>
                  ))
                }
              </div>
              <h3>{show.premiered}</h3>
              <h3>Rating: {show.rating.average}</h3>
              <h3>Language: {show.language}</h3>
              <div dangerouslySetInnerHTML={{__html: show.summary}} />
              { this.renderEpisodes() }
            </Col>
            <Col xs={{ span: 6, pull: 16 }} sm={{ span: 6, pull: 16 }} md={{ span: 6, pull: 16 }} lg={{ span: 6, pull: 16 }} >
              <Card
                hoverable
                cover={<img alt='example' src={show.image && show.image.medium ? show.image.medium : 'https://placeimg.com/210/295/any'} />}
              />
            </Col>
          </Row>
        </NavigateApp>
      )
    }
    return this.renderLoader(config)
  }
}

const mapDispatchToProps = (dispatch) => ({
  getOneShow: (id) => dispatch(actions.getOneShow(id)),
  getShowEpisodes: (id) => dispatch(actionsEpisodes.getShowEpisodes(id))
})

const mapStateToProps = state => {
  const { shows, episodes } = state
  return {
    shows,
    episodes
  }
}

export default withRedux(makeStore, mapStateToProps, mapDispatchToProps)(DetailsConatiner)
