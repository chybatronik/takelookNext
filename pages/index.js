import React, { Component } from 'react'
import * as actions from '../store/actions/shows'
import { connect } from 'react-redux'
import NavigateApp from '../components/Layout.js'
import { Row, Col, Spin, Alert } from 'antd'
import InfiniteScroll from 'react-infinite-scroller'
import ShowComponent from '../components/show'
import { makeStore } from '../store'
import Router from 'next/router'
import withRedux from "next-redux-wrapper";

export class DefaultListContainer extends Component {
  static async getInitialProps (params) {
    console.log('getInitialProps DefaultListContainer')
    const {query, store, isServer} = params
    await store.dispatch(actions.getShows(0))
    return store.getState()
  }

  renderLoader () {
    const { config } = this.props.shows
    if (config.isError) {
      return (
        <Alert message='Ошибка получения данных с сервера' type='error' />
      )
    }
    return (
      <div className='spiner' key={'spiner'}>
        <Spin size='large' />
      </div>
    )
  }
  handleOnClick = (id) => {
    Router.push({
     pathname: '/details',
     query: { id: id }
   })
  }

  loadFunc = () => {
    const { config } = this.props.shows
    if(!config.isFetch)
      this.props.getShows(config.page)
  }

  render () {
    const { shows } = this.props.shows
    const results = Object.keys(shows).map((item) => {
      return (
        <Col xs={{ span: 12 }} sm={{ span: 8 }} md={{ span: 6 }}  lg={{ span: 4 }} style={{paddingBottom: 20}} key={item}>
          <ShowComponent handleOnClick={this.handleOnClick} show={shows[item]} key={item} />
        </Col>
      )
    })
    return (
      <NavigateApp title={"All shows"}>
        <InfiniteScroll
          pageStart={0}
          loadMore={this.loadFunc}
          hasMore={true}
          loader={this.renderLoader()}
        >
          <Row gutter={16}>
            {results}
          </Row>
        </InfiniteScroll>
      </NavigateApp>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  getShows: (page) => dispatch(actions.getShows(page))
})

const mapStateToProps = state => {
  const { shows } = state
  return {
    shows
  }
}


export default withRedux(makeStore, mapStateToProps, mapDispatchToProps)(DefaultListContainer)
