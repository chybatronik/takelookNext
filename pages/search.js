import React, { Component } from 'react'
import * as actions from '../store/actions/shows'
// import { connect } from 'react-redux'
import { Row, Col, Spin, Alert } from 'antd'
// import InfiniteScroll from 'react-infinite-scroller'
import ShowComponent from '../components/show'
// import withRedux from '../utils/withRedux'
import { makeStore } from '../store'
import NavigateApp from '../components/Layout.js'
import Router from 'next/router'
import withRedux from 'next-redux-wrapper'

export class SearchContainer extends Component {
  static async getInitialProps (params) {
    const {query, store, isServer} = params
    console.log('getInitialProps SearchContainer', query.name)
    // await store.dispatch(actions.getShows(0))
    await store.dispatch(actions.getSearchShows(query.name))
    return {...store.getState(), queryName: query.name}
  }

  // componentDidMount () {
  //   const { config } = this.props.shows
  //   if(!config.isFetch)
  //     this.props.getSearchShows(Router.query.name)
  // }
  //
  // componentWillUpdate (nextProps, nextState) {
  //   if (nextProps.url !== this.props.url) {
  //     if(!nextProps.shows.config.isFetch)
  //       this.props.getSearchShows(nextProps.url)
  //   }
  // }

  renderLoader () {
    const { config } = this.props.shows
    if (config.isError) {
      return (
        <NavigateApp title={'Ошибка '}>
          <Alert message='Ошибка получения данных с сервера' type='error' />
        </NavigateApp>
      )
    }
    return (
      <NavigateApp title={'Ошибка '}>
        <div className='spiner' key={'spiner'}>
          <Spin size='large' />
        </div>
      </NavigateApp>
    )
  }

  handleOnClick = ( id ) => {
    Router.push({
     pathname: '/details',
     query: { id: id }
   })
  }

  render () {
    const { shows, config } = this.props.shows
    if (config.isFetch || config.isError) {
      return this.renderLoader(config)
    }
    const results = Object.keys(shows).map((item) => {
      return (
        <Col xs={{ span: 12 }} sm={{ span: 8 }} md={{ span: 6 }}  lg={{ span: 4 }} style={{paddingBottom: 20}} key={item}>
          <ShowComponent handleOnClick={this.handleOnClick} show={shows[item]} key={item} />
        </Col>
      )
    })

    return (
      <NavigateApp title={'Вы искали ' + this.props.queryName}>
        <Row gutter={16}>
          <h2>Вы искали: {this.props.queryName}</h2>
        </Row>
        <Row gutter={16}>
          {results}
        </Row>
      </NavigateApp>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  getSearchShows: (query) => dispatch(actions.getSearchShows(query))
})

const mapStateToProps = state => {
  const { shows } = state
  return {
    shows
  }
}

export default withRedux(makeStore, mapStateToProps, mapDispatchToProps)(SearchContainer)
