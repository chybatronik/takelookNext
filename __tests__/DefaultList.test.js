/**
 * @jest-environment node
 */
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import expect from 'expect'
import { DefaultListContainer } from '../pages/index'

// Enzyme.configure({ adapter: new Adapter() })

it('DefaultListContainer isFetch false', () => {
  const config = {isFetch: false}
  const shows = {
    shows: {},
    config
  }
  const output = shallow(
    <DefaultListContainer shows={shows} />
  )
  expect(shallowToJson(output)).toMatchSnapshot()
})

it('DefaultListContainer isFetch true', () => {
  const config = {isFetch: true}
  const shows = {
    shows: {},
    config
  }
  const output = shallow(
    <DefaultListContainer shows={shows} />
  )
  expect(shallowToJson(output)).toMatchSnapshot()
})
