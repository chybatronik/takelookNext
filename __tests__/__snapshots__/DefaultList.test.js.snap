// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultListContainer isFetch false 1`] = `
<Component
  title="All shows"
>
  <InfiniteScroll
    element="div"
    hasMore={true}
    initialLoad={true}
    isReverse={false}
    loadMore={[Function]}
    loader={
      <div
        className="spiner"
      >
        <Spin
          prefixCls="ant-spin"
          size="large"
          spinning={true}
          wrapperClassName=""
        />
      </div>
    }
    pageStart={0}
    ref={null}
    threshold={250}
    useCapture={false}
    useWindow={true}
  >
    <Row
      gutter={16}
    />
  </InfiniteScroll>
</Component>
`;

exports[`DefaultListContainer isFetch true 1`] = `
<Component
  title="All shows"
>
  <InfiniteScroll
    element="div"
    hasMore={true}
    initialLoad={true}
    isReverse={false}
    loadMore={[Function]}
    loader={
      <div
        className="spiner"
      >
        <Spin
          prefixCls="ant-spin"
          size="large"
          spinning={true}
          wrapperClassName=""
        />
      </div>
    }
    pageStart={0}
    ref={null}
    threshold={250}
    useCapture={false}
    useWindow={true}
  >
    <Row
      gutter={16}
    />
  </InfiniteScroll>
</Component>
`;
